File: backoffice/src/diningTable/domain/valueObjects/DiningTableId.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain                          import UuidValueObject
from ..exceptions.InvalidDiningTableIdException import InvalidDiningTableIdException

"""
 *
 * Class
 *
"""

class DiningTableId( UuidValueObject ):

    """
     *
     * Methods 
     *
    """
    
    def __init__( self, value : str ) -> None:
        super().__init__( value )
        if self.isEmpty():
            raise InvalidDiningTableIdException( value )

File: preparation/src/diningTable/domain/valueObjects/DiningTableId.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.diningTable.domain {

    public class DiningTableId : UuidValueObject {

        /* 
         * 
         * Methods
         *
        */

        public DiningTableId( string value ) : base( value ) {
            if( isEmpty() ) {
                throw new InvalidDiningTableIdException( value );
            }
        }

    }

}